test/integration/src/main/java/io/pravega/test/integration/compatibility/CompatibilityChecker.java

```diff
@@ -0,0 +1,225 @@
+ * limitations under the License.
+ */
+package io.pravega.test.integration.compatibility;
+
+import io.pravega.client.ClientConfig;
+import io.pravega.client.EventStreamClientFactory;
+import io.pravega.client.admin.ReaderGroupManager;
+import io.pravega.client.admin.StreamManager;
+import io.pravega.client.stream.DeleteScopeFailedException;
+import io.pravega.client.stream.EventRead;
+import io.pravega.client.stream.EventStreamReader;
+import io.pravega.client.stream.EventStreamWriter;
+import io.pravega.client.stream.EventWriterConfig;
+import io.pravega.client.stream.ReaderConfig;
+import io.pravega.client.stream.ReaderGroup;
+import io.pravega.client.stream.ReaderGroupConfig;
+import io.pravega.client.stream.ReinitializationRequiredException;
+import io.pravega.client.stream.ScalingPolicy;
+import io.pravega.client.stream.Stream;
+import io.pravega.client.stream.StreamConfiguration;
+import io.pravega.client.stream.StreamCut;
+import io.pravega.client.stream.impl.UTF8StringSerializer;
+import lombok.Cleanup;
+import lombok.extern.slf4j.Slf4j;
+
+import java.net.URI;
+import java.util.UUID;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@Slf4j
+public class CompatibilityChecker {
+    private static final int READER_TIMEOUT_MS = 2000;
+    public  URI controllerURI;
+    public  StreamManager streamManager;
+    public StreamConfiguration streamConfig;
+
+    public  void setUp() {
+        controllerURI = URI.create("tcp://localhost:9090");
+        streamManager = StreamManager.create(controllerURI);
+        streamConfig = StreamConfiguration.builder().scalingPolicy(ScalingPolicy.fixed(1)).build();
+    }
+
+    /**
+    * This method is Checking the working of the read and write event of the stream.
+    * Here we are trying to create a stream and a scope, and then we are writing a couple of events.
+    * And then reading those written event from the stream.
+    */
+    public void checkWriteAndReadEvent() {
+        String scopeName = "write-and-read-test-scope";
+        String streamName = "write-and-read-test-stream";
+        streamManager.createScope(scopeName);
+        streamManager.createStream(scopeName, streamName, streamConfig);
+        @Cleanup
+        EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(scopeName, ClientConfig.builder().controllerURI(controllerURI).build());
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(streamName, new UTF8StringSerializer(), EventWriterConfig.builder().build());
+        String readerGroupId = UUID.randomUUID().toString().replace("-", "");
+        ReaderGroupConfig readerGroupConfig = ReaderGroupConfig.builder().stream(Stream.of(scopeName, streamName)).build();
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scopeName, controllerURI);
```

Comment:
Please, close this resource once used (add `Cleanup`). Also, look for any other resource in this code that may need to be closed and it is not.

Reply:
done